### ISSUE TITLE
2.13 - Accéder aux listes de JDD et CA sans accéder au module Métadonnées ?

### ISSUE THREAD

**Issue by camillemonchicourt** [PnX-SI/GeoNature#2714]
Dans plusieurs modules on a besoin de lister des JDD : 
- Dans les modules de saisie comme Occtax, Occhab, Import... pour pouvoir associer les données à un JDD
- Dans la Synthèse pour filtrer par JDD ou CA

On utilise pour cela la route `/datasets` qui est aussi utilisée pour lister les JDD dans le module Métadonnées.

Dans les modules de saisie, il a été fait en sorte (dans la 2.9.0) de pouvoir limiter la liste des JDD en fonction de la portée de l'action C du module (cela permet ainsi d'accéder à tous les JDD dans le module Métadonnées ou dans la Synthèse, mais de ne pouvoir associer des relevés Occtax qu'aux JDD de son organisme), comme détaillé ici - https://github.com/PnX-SI/GeoNature/issues/659#issuecomment-972829195.

Pour être cohérent et ne pas pouvoir accéder à des JDD auxquels on ne devrait pas avoir accès, on compare la portée globale de lecture sur les métadonnées et la portée sur le C du module de saisie pour ne pas renvoyer dans la module de saisie des JDD auxquels on n'a pas le droit d'accéder globalement.

Dans les modules avec de la recherche par JDD ou CA (comme le module Synthèse), on ne regarde que les permissions globales sur les métadonnées pour les lister dans les filtres.

Jusqu'à la version 2.13, ces permissions globales étaient celles définies globalement au niveau de GeoNature.
Avec la 2.13 où l'héritage des permissions ont été supprimées, il n'y a plus de permissions globales sur GeoNature, mais uniquement des permissions explicites définies module par module.
La conséquence est que pour lister les JDD et CA dans les filtres de recherche des modules (comme la Synthèse) ou dans les modules de saisie, on regarde les permissions et la portée de l'action R du module Métadonnées (croisées avec celles de l'action C du module pour les modules de saisie).
Cela est fait et expliqué dans cette PR : https://github.com/PnX-SI/GeoNature/pull/2467

La conséquence principale de cela (mentionnée dans cette même PR) est que pour pouvoir avoir des JDD et CA listés dans les modules de saisie ou dans les filtres de la Synthèse, il faut des droits R dans le module Métadonnées. 
Et donc on ne peut plus avoir des CA et JDD dans ces listes, sans avoir accès au module Métadonnées.

Sinon les listes des CA et JDD sont vides et on a un message d'erreur quand on arrive dans un module qui cherche à lister des JDD : 

![image](https://github.com/PnX-SI/GeoNature/assets/4418840/f94ec563-ffac-429d-9635-4c7076da4ddf)

C'est assez logique que si on peut voir une liste de JDD, alors on peut en consulter le détail dans le module Métadonnées.

Mais certains veulent donner accès aux modules Synthèse et/ou Occtax sans forcément donner accès au module Métadonnées, comme dans ce ticket - https://github.com/PnX-SI/GeoNature/issues/1841#issuecomment-1120925911

Cela n'est plus possible avec la suppression de l’héritage des permissions en 2.13.

A voir comment gérer ce besoin.

-----------------------------

Cela est lié au fait que la logique des permissions de GeoNature mélange des permissions sur des objets/entités globales et transversales (comme les métadonnées) et l'accès ou non aux modules, ce qui est parfois contradictoire, limitant ou confusant.

On pourrait imaginer ne lire que le R de Synthèse pour lister les CA et JDD dans les filtres de la Synthèse, mais cela permettrait d'accéder au détail des métadonnées dans l'API même en aillant aucun R sur le module Métadonnées, donc pas souhaitable non plus.

---------------------------

On peut aussi considérer que c'est normal et souhaitable et que le module Métadonnées est central et que vu qu'on a accès à des listes de JDD et CA, autant pouvoir accès à leur détail dans le module Métadonnées.
Dans les fiches infos des observations de la Synthèse, on indique leur CA et JDD. On pourrait ainsi les rendre cliquable pour accéder à leur détail dans le module Métadonnées.

**Comment by gildeluermoz** [comment]
Oui, il est logique qu'il faille une permission sur de Read sur les metadonnées pour pouvoir les voir dans les listes déroulantes des autres modules.
Comme indiqué dans ce ticket : https://github.com/PnX-SI/GeoNature/issues/4058, ce qui est moins intuitif, c'est qu'un filtre de portée sur l'action Create dans un module, interfère avec le contenu de la liste déroulante des metadonnées, même si il y a un Read sur "Voir les métadonnées".
Dans ma logique, qui n'est que ma logique, un Create conduit à créer un enregistrement et cette action n'a pas de portée, ou plus précisément, cette action a une portée qui est toujours globale.
